miner/src/pool/replace.rs

```diff
@@ -71,6 +71,13 @@ where
 			let old_score = (old.priority(), old.gas_price());
 			let new_score = (new.priority(), new.gas_price());
 			if new_score > old_score {
+				// check for an existing transaction with the same nonce
+				if let Some(txs) = new.pooled_by_sender {
+					if let Ok(index) = txs.binary_search_by(|old| self.scoring.compare(old, new)) {
+						return self.scoring.choose(&txs[index], new)
```

Comment:
We should convert `ReplaceOld` to `InsertNew`, to prevent removing the worst transaction from the pool - since we detected that this is a replacement transaction, it will, well, replace some other transaction anyway.